gobgp/cmd/global.go

```diff
@@ -809,8 +811,21 @@ func ParsePath(rf bgp.RouteFamily, args []string) (*table.Path, error) {
 		if err != nil {
 			return nil, err
 		}
-		p := bgp.NewPathAttributeExtendedCommunities(extcomms)
+		normalextcomms := make([]bgp.ExtendedCommunityInterface, 0)
+		ipv6extcomms := make([]bgp.ExtendedCommunityInterface, 0)
+		for _, com := range extcomms {
+			bstring, _ := com.Serialize()
+			switch len(bstring) {
+			case 20:
+				ipv6extcomms = append(ipv6extcomms, com)
+			default:
+				normalextcomms = append(normalextcomms, com)
+			}
+		}
+		p := bgp.NewPathAttributeExtendedCommunities(normalextcomms)
 		attrs = append(attrs, p)
+		ip6p := bgp.NewPathAttributeIP6ExtendedCommunities(ipv6extcomms)
```

Comment:
This always adds an IPv6 extended community path attribute even if `len(ipv6extcomms) == 0`. We need to check both `len(normalextcomms)` and `len(ipv6extcomms)`.